R/url_path.R

```diff
@@ -0,0 +1,20 @@
+#' Construct URL Path
+#' 
+#' Construct a URL from path components like \code{file.path}
+#'
+#' @param x a URL, or vector of URLs
+#' @param ... character vectors
+#' 
+#' @examples
+#' 
+#' url_path("https://github.com", c('Ironholds', 'nfultz'), 'urltools')
+#' 
+#' @export
+#'
+url_path <- function(x, ...) {
+  p <- path(x)
+  p[is.na(p)] = '.'
```

Comment:
Correct. If the one-liner: `path(x) <- file.path(path(x), foo, bar, baz)` didn't have problems with NAs, leading slashes, and vectorization, I would rather just write that since it's more obvious. At the most, it might be a good example in the path help page.